test/test_view.c

```diff
@@ -185,13 +189,72 @@ int main(int argc, char *argv[])
             fprintf(stderr, "[E::%s] fail to load the BAM index\n", __func__);
             return 1;
         }
-        for (i = optind + 1; i < argc; ++i) {
-            hts_itr_t *iter;
-            if ((iter = sam_itr_querys(idx, h, argv[i])) == 0) {
-                fprintf(stderr, "[E::%s] fail to parse region '%s'\n", __func__, argv[i]);
-                continue;
+        if (multi_reg) {
+            int reg_count = 0;
+            hts_reglist_t *reg_list = calloc(argc-(optind+1), sizeof(*reg_list));
+            if (!reg_list)
+                return 1;
+
+            // We need a public function somewhere to turn an array of region strings
+            // into a region list, but for testing this will suffice for now.
+            // Consider moving a derivation of this into htslib proper sometime.
+            for (i = optind + 1; i < argc; ++i) {
+                int j;
+                uint32_t beg, end;
+                char *cp = strrchr(argv[i], ':');
+                if (cp) *cp = 0;
+
+                // Does any of this need to be sorted?
+                // For now we assume any such limits are passed onto whoever
+                // is running the command line.
+                for (j = 0; j < reg_count; j++)
+                    if (strcmp(reg_list[j].reg, argv[i]) == 0)
+                        break;
+                if (j == reg_count) {
+                    reg_list[j=reg_count++].reg = argv[i];
```

Comment:
`j=reg_count` part is a bit redundant here, since `j` was already equal to `reg_count`.